### PR TITLE
added unique parameter to members that needed it for db

### LINKED ISSUE #77: Add unique constraints to the DB

### PR DESCRIPTION
**Pull request by owenjaques**
# Description
This PR is pretty small as it just adds unique keys to fields I felt needed it in the DB. 

Fixes #77 
## Type of change
Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
# How Has This Been Tested?
- [x] Ran docker with new changes and it worked
# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have linted my code and fixed all issues

### CODE REVIEW

**Comment by owenjaques** [comment]
Ahhh test is failing now, please don't merge this. I will figure out what is going on.

**Review by itsazoo — CHANGES_REQUESTED** [review]
Is it possible to have a unique constaint on a pair? For ProfessorAvailability, profs should only have one entry per year. Therefore (prof_id, year) should be a unique pair. Similarly, in ProfessorCoursePreference (course_id, prof_avail_id) should be a unique pair. 
Everything else looks good!

**Comment by itsazoo** [comment]
> Is it possible to have a unique constaint on a pair? For ProfessorAvailability, profs should only have one entry per year. Therefore (prof_id, year) should be a unique pair. Similarly, in ProfessorCoursePreference (course_id, prof_avail_id) should be a unique pair.
> Everything else looks good!
Looks like this might do the trick. (https://stackoverflow.com/questions/635937/how-do-i-specify-unique-constraint-for-multiple-columns-in-mysql)
`UNIQUE KEY `my_uniq_id` (`course_id `,`prof_avail_id `)`

File: database/sql/schema.sql [hunk 1]
```diff
@@ -39,8 +39,8 @@ CREATE TABLE HistoricalData(
 
 CREATE TABLE CourseOffering (
 	id BINARY(16) PRIMARY KEY,
-	course_name VARCHAR(100) NOT NULL,
-	course_code VARCHAR(10) NOT NULL, 
+	course_name VARCHAR(100) NOT NULL UNIQUE,
+	course_code VARCHAR(10) NOT NULL UNIQUE, 
 	course_desc VARCHAR(2000) NOT NULL,
 	prof_prereq VARCHAR(500),
 	min_offering INT NOT NULL,
```
Comment (itsazoo) [inline database/sql/schema.sql:46]: We now support saving historical preferences for profs therefore we need to add a unique constraint in ProfessorAvailability table: 

- [ ] Add `UNIQUE KEY `my_uniq_id` (`year`,`prof_id`)` to ProfessorAvilability table

**Review by itsazoo — CHANGES_REQUESTED** [review]

**Comment by itsazoo** [comment]
Are we still planning to add this functionality?

**Comment by ekanemeinstein** [comment]
Out of scope

**Pull request closed without merging**